modules/build/src/main/scala/scala/build/bsp/BspImpl.scala

```diff
@@ -287,10 +293,37 @@ final class BspImpl(
 
     preBuild.thenCompose {
       case Left((ex, scope)) =>
+        val targetId  = currentBloopSession.bspServer.targetIds.head
```

Comment:
We need to handle the case when `buildTargetId` doesn't exist and when there are several `targetIds`:
```suggestion
        for (targetId <- currentBloopSession.bspServer.targetIds) {
          ...
        }
```